ospfd/ospf_sr.c

```diff
@@ -1115,38 +1149,85 @@ void ospf_sr_ri_lsa_update(struct ospf_lsa *lsa)
 		}
 	}
 
-	/* Check that we collect mandatory parameters */
-	if (srn->algo[0] == SR_ALGORITHM_UNSET || srgb.range_size == 0
-	    || srgb.lower_bound == 0) {
-		flog_err(EC_OSPF_SR_NODE_CREATE,
-			 "SR (%s): Missing mandatory parameters. Abort!",
-			 __func__);
-		hash_release(OspfSR.neighbors, &(srn->adv_router));
-		XFREE(MTYPE_OSPF_SR_PARAMS, srn);
+	/* Check if Segment Routing Capabilities has been found */
+	if (ri_srgb == NULL) {
+		/* Skip Router Information without SR capabilities
+		 * advertise by a non SR Node */
+		if (srn == NULL) {
+			return;
+		} else {
+			/* Remove SR Node that advertise Router Information
+			 * without SR capabilities. This could correspond to a
+			 * Node stopping Segment Routing */
+			hash_release(OspfSR.neighbors, &(srn->adv_router));
+			sr_node_del(srn);
+			return;
+		}
+	}
+
+	/* Check that RI LSA belongs to the correct SR Node */
+	if ((srn != NULL) && (srn->instance != 0)
+	    && (srn->instance != ntohl(lsah->id.s_addr))) {
+		flog_err(EC_OSPF_SR_INVALID_LSA_ID,
+			 "SR (%s): Abort! Wrong "
+			 "LSA ID 4.0.0.%u for SR node %pI4/%u",
+			 __func__, GET_OPAQUE_ID(ntohl(lsah->id.s_addr)),
+			 &lsah->adv_router, srn->instance);
 		return;
 	}
 
+	/* OK. All things look good. Get SRGB */
+	srgb.range_size = GET_RANGE_SIZE(ntohl(ri_srgb->size));
+	srgb.lower_bound = GET_LABEL(ntohl(ri_srgb->lower.value));
+
 	/* Check if it is a new SR Node or not */
-	if (srn->instance == 0) {
+	if (srn == NULL) {
+		/* Get a new SR Node in hash table from Router ID */
+		srn = (struct sr_node *)hash_get(OspfSR.neighbors,
+						 (void *)&(lsah->adv_router),
+						 (void *)sr_node_new);
```

Comment:
nit: unnecessary explicit casts to `void *` and unnecessary parentheses around `lsah->adv_router`

Reply:
done only for lsah->adv_router as compiler hangs a warning if I suppress the (void *) for sr_node_new